options/options.go

```diff
@@ -44,7 +44,10 @@ var (
 )
 
 const IncompatibleArgsErrorFormat = "illegal argument combination: cannot specify %s and --uri"
-const ConflictingArgsErrorFormat = "illegal argument combination: %s conflicts with --uri"
+
+func ConflictingArgsErrorFormat(optionName, uriValue, cliValue, cliOptionName string) error {
+	return fmt.Errorf("Invalid Options: Cannot specify different %s in connection URI and command-line option (\"%s\" was specified in the URI and \"%s\" was specified in the %s option)", optionName, uriValue, cliValue, cliOptionName)
+}
```

Comment:
It might be beneficial to keep a mapping from cliOptionName to optionName so that you don't need to pass both arguments, but I'm fine with this.